src/jit/compiler.cpp

```diff
@@ -2478,6 +2479,28 @@ void Compiler::compSetProcessor()
 #endif // _TARGET_AMD64_
             }
         }
+#else  // !FEATURE_CORECLR
+        // If this is not FEATURE_CORECLR, the flags for SSE3, SSSE3, SSE41 and SSE42 are collectively
+        // implied by EnableSSE3_4, and, since we don't support the HW intrinsics, these and the AVX
+        // ISAs are the only ones supported by the VM and relied upon by the JIT.
+        //
+        if (JitConfig.EnableSSE3_4())
+        {
+            opts.setSupportedISA(InstructionSet_SSE3);
```

Comment:
Don't forget `SSSE3`

Reply:
Digging into this a bit further, it seems that the desktop the only ISA-related flags supported by the VM are `CORJIT_FLAG_USE_AVX` and `CORJIT_FLAG_USE_AVX2`. So I think this can and should be further simplified. Apparently the code that utilizes the in-between ISAs was added after the VM sources were split off.